sopt30th_1st HW/sopt30th_1st HW/RegisterIDViewController.swift

```diff
@@ -0,0 +1,53 @@
+//
+//  RegisterIDViewController.swift
+//  sopt30th_1st HW
+//
+//  Created by 최영린 on 2022/04/06.
+//
+
+import UIKit
+
+class RegisterIDViewController: UIViewController {
+
+    @IBOutlet weak var idTextField: UITextField!
+    @IBOutlet weak var nextButton: UIButton! {
+        didSet{
+            nextButton.isEnabled = false
+        }
+    }
+    
+    var user: String?
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        setBackButton()
+        setNextButton()
+    }
+    
+
+    @IBAction func idNextButtonDidTap(_ sender: Any) {
+        guard let registerPWVC = self.storyboard?.instantiateViewController(withIdentifier: "RegisterPWViewController") as? RegisterPWViewController else { return }
+        
+        registerPWVC.user = idTextField.text
+                
+        self.navigationController?.pushViewController(registerPWVC, animated: true)
+    }
+    
+    private func setBackButton() {
+        let backBarButtonItem = UIBarButtonItem(title: "", style: .plain, target: nil, action: nil)
+                navigationItem.backBarButtonItem = backBarButtonItem
+    }
+    
+    private func textHandler(_ a: UIAction) -> Void {
+        if self.idTextField.text?.isEmpty == true {
+            self.nextButton.isEnabled = false
+        } else {
+            self.nextButton.isEnabled = true
+        }
+    }
```

Comment:
이 부분에서는 코드를 줄일 수 있겠네용~!
self.idTextField.text?.isEmpty가 이미 Boolean 값이기 때문에
```suggestion
    private func textHandler(_ a: UIAction) -> Void {
        if self.idTextField.text?.isEmpty {
            self.nextButton.isEnabled = false
        } else {
            self.nextButton.isEnabled = true
        }
    }
```

이런 식으로 해주는 것도 좋을 것 같아요~!

Reply:
또 저는 명확하게 상황을 알려주는 코드도 읽기 좋아서 너무 좋아하지만,, 간결한 코드도 좋아하는 사람으로서,,
Boolean 값들만의 처리를 할 땐 직접 대입을 해주는 것도
코드도 짧아지고, 이미 충분히 상황을 파악할 수 있게 짧아질 수 있을 거라고 생각해요 ~!

지금 이 경우엔 idTextField가 비어있다면?(true) -> 버튼의 활성화는 안된다(false)
정반대 값을 가지고 있으니까 !값으로 쇽샥 바꿔줄 수 잇갯내용~!

```suggestion
    private func textHandler(_ a: UIAction) -> Void {
        self.nextButton.isEnabled = !(self.idTextField.text?.isEmpty)
    }
```

이런 식으로요!

Reply:
코드 간결화를 위해 문법공부가 필요함을 몸소 느끼고 있어요! 자세히 알려주셔서 정말 감사합니다!!